src/core/mac/mac.cpp

```diff
@@ -1740,7 +1740,7 @@ void Mac::HandleReceivedFrame(RxFrame *aFrame, Error aError)
 
     IgnoreError(aFrame->GetSrcAddr(srcaddr));
     IgnoreError(aFrame->GetDstAddr(dstaddr));
-    neighbor = Get<NeighborTable>().FindNeighbor(srcaddr);
+    neighbor = !srcaddr.IsNone() ? Get<NeighborTable>().FindNeighbor(srcaddr) : nullptr;
```

Comment:
Thanks @trebbert-lutron for finding and fixing this.

I think it may be better if we fix this in `FindNeighbor()`. I would say it should return `nullptr` if `Mac::Address` is empty/none (which is a valid state of `Mac::Address`). This way we fix in all cases that `FindNeighbor()` may be used.

Reply:
That makes sense to me - assuming that's the desired behavior in all cases. I wasn't sure if there was a use case where that would break somewhere within the stack that I don't know about. I also saw that this pattern was used in other places, for example: https://github.com/openthread/openthread/blob/main/src/core/mac/mac.cpp#L1659-L1670

But if you can be confident that making the change within FindNeighbor() makes the most sense, I'm on board.

Reply:
I am looking at the code and different situation related to this.

There are actually cases where `Neighbor::AddressMatcher` should treat "no address"  as "accept any address" (e.g., this seems to be used to find new (unused) child).

Let me look a bit more into this. If I find a good way I will submit a PR.

Reply:
After looking more, I think it may be better to not to touch the `AddressMatcher` logic.
I agree checking this at `Mac` seems to be best way. Thanks.